### PR TITLE
Minor changes to proto generation

### PR DESCRIPTION
**Pull request by mooskagh**
* `reserved` keyword is now supported.
    * It's possible to reserve fields by number, or by name (quotes are optional).
* string fields now have `mutable_X()` function.

### CODE REVIEW

**Comment by borg323** [comment]
Is it hard to make it accept the `reserved 10 to 18;` syntax?

**Comment by mooskagh** [comment]
Done (didn't test, should work)

File: scripts/compile_proto.py [hunk 1]
```diff
@@ -630,6 +637,10 @@ def CheckReserved(self):
             if isinstance(r, int):
                 if any(x.number == r for x in self.fields):
                     raise ValueError(f'Field number [{r}] is reserved.')
+            elif isinstance(r, range):
+                if any(x.number in r for x in self.fields):
+                    raise ValueError(f'Field range [{r.start} to {r.stop}] '
```
Comment (borg323) [inline scripts/compile_proto.py:642]: I think this should be `r.stop - 1`.

Comment (mooskagh) [reply scripts/compile_proto.py]: Indeed, done!

**Review by borg323 — COMMENTED** [review]
I did some testing of `reserved`, looks good.

**Review by borg323 — APPROVED** [review]
Approving again after rebase.

**Pull request merged**